### PR TITLE
Lease Notes | psp-4807

### PR DESCRIPTION
**Pull request by FuriousLlama**
Added the note tab to the frontend leases and wired up backend

### CODE REVIEW

File: source/backend/api/Services/LeaseService.cs [hunk 1]
```diff
@@ -84,7 +111,7 @@ private PimsLease AssociatePropertyLeases(PimsLease lease, bool userOverride = f
         {
             MatchProperties(lease);
 
-            lease.PimsPropertyLeases.ForEach(propertyLease =>
+            foreach (var propertyLease in lease.PimsPropertyLeases)
```
Comment (devinleighsmith) [inline source/backend/api/Services/LeaseService.cs:114]: What is the benefit of making this change?

File: source/backend/api/Services/LeaseService.cs [hunk 2]
```diff
@@ -64,6 +71,26 @@ public PimsLease Add(PimsLease lease, bool userOverride = false)
...
         {
+            var currentLease = _leaseRepository.Get(lease.LeaseId);
+
+            if (currentLease.LeaseStatusTypeCode != lease.LeaseStatusTypeCode)
+            {
+                _entityNoteRepository.Add<PimsLeaseNote>(
+                    new PimsLeaseNote()
+                    {
+                        LeaseId = currentLease.LeaseId,
+                        AppCreateTimestamp = System.DateTime.Now,
```
Comment (devinleighsmith) [inline source/backend/api/Services/LeaseService.cs:82]: is this necessary? I know we do some automatic insertion of these audit values in most scenarios. Not sure if that is applicable here.

File: source/backend/api/Services/NoteService.cs [hunk 3]
```diff
@@ -58,21 +58,29 @@ public EntityNoteModel Add(NoteType type, EntityNoteModel model)
             switch (type)
             {
                 case NoteType.Activity:
-                    var pimsEntity = _mapper.Map<PimsActivityInstanceNote>(model);
+                    PimsActivityInstanceNote pimsEntity = _mapper.Map<PimsActivityInstanceNote>(model);
```
Comment (devinleighsmith) [inline source/backend/api/Services/NoteService.cs:61]: Should we talk about best practices around usage of var?

File: source/backend/api/Services/LeaseService.cs (hunk 1 above)
Comment (FuriousLlama) [reply source/backend/api/Services/LeaseService.cs:114]: The ForEach was conflicting with the LINQ namespace and the extensions we have

File: source/backend/dal/Repositories/EntityNoteRepository.cs [hunk 4]
```diff
@@ -100,6 +109,24 @@ public bool DeleteAcquisitionFileNotes(long entityId)
...
+        {
+            var leaseFileNotes = this.Context.PimsLeaseNotes.
+                                        Include(ln => ln.Note).
+                                        Where(x => x.NoteId == noteId).ToList();
+
+            if (leaseFileNotes.Any())
+            {
+                foreach (var leaseFileNote in leaseFileNotes)
+                {
+                    this.Context.PimsLeaseNotes.Remove(leaseFileNote);
```
Comment (devinleighsmith) [inline source/backend/dal/Repositories/EntityNoteRepository.cs:122]: I wonder if we need this now or if our cascades or covering this. Maybe we need to go back now that we have some cascade logic and clean some of these up? certainly not something you need to worry about this PR but I should probably log tech debt for it.

File: source/backend/dal/Repositories/Interfaces/IEntityNoteRepository.cs [hunk 5]
```diff
@@ -14,10 +14,14 @@ IEnumerable<T> GetAll<T>(Func<T, bool> predicate)
 
         IEnumerable<PimsNote> GetAllActivityNotesById(long entityId);
 
-        IEnumerable<PimsNote> GetAllAcquisitionNotesById(long entityId);
+        IEnumerable<PimsNote> GetAllAcquisitionNotesById(long acquisitionId);
```
Comment (devinleighsmith) [inline source/backend/dal/Repositories/Interfaces/IEntityNoteRepository.cs:17]: I like that you've made this more readable but kinda surprised you just targeted AcqNotes?

**Review by devinleighsmith — APPROVED** [review]

File: source/backend/dal/Repositories/Interfaces/IEntityNoteRepository.cs (hunk 5 above)
Comment (FuriousLlama) [reply source/backend/dal/Repositories/Interfaces/IEntityNoteRepository.cs:17]: Didn't want to update Activity related code :)

Comment (devinleighsmith) [reply source/backend/dal/Repositories/Interfaces/IEntityNoteRepository.cs:17]: lol, makes sense.

**Pull request merged**